scene/gui/code_edit.cpp

```diff
@@ -1419,41 +1419,27 @@ void CodeEdit::fold_line(int p_line) {
 		/* End line is the same therefore we have a block of single line delimiters. */
 		if (end_line == p_line) {
 			for (int i = p_line + 1; i <= line_count; i++) {
-				if (i == line_count) {
-					end_line = line_count;
+				if ((in_string != -1 && is_in_string(i) == -1) || (in_comment != -1 && is_in_comment(i) == -1)) {
+					end_line = i - 1;
 					break;
 				}
 
-				if ((in_string != -1 && is_in_string(i) == -1) || (in_comment != -1 && is_in_comment(i) == -1)) {
-					end_line = i - 1;
+				if (i == line_count) {
+					end_line = line_count;
 					break;
 				}
 			}
 		}
 	} else {
 		int start_indent = get_indent_level(p_line);
 		for (int i = p_line + 1; i <= line_count; i++) {
-			if (get_line(p_line).strip_edges().size() == 0 || is_in_string(i) != -1 || is_in_comment(i) != -1) {
-				end_line = i;
-				continue;
-			}
-
-			if (i == line_count) {
-				/* Do not fold empty last line of script if any */
-				end_line = i;
-				if (get_line(i).strip_edges().size() == 0) {
-					end_line--;
-				}
-				break;
-			}
-
-			if ((get_indent_level(i) <= start_indent && get_line(i).strip_edges().size() != 0)) {
-				/* Keep an empty line unfolded if any */
-				end_line = i - 1;
-				if (get_line(i - 1).strip_edges().size() == 0 && i - 2 > p_line) {
-					end_line = i - 2;
+			if (get_line(i).strip_edges().size() != 0) {
```

Comment:
Nit: Can invert this

Reply:
Could you clarify what you mean by this please?

Reply:
Sure, invert the condition so it becomes:
```
...
if (get_line(i).strip_edges().size() == 0) {
    continue;
}
...
```

Reply:
Thanks, now added  :)